src/Mapper/ProgrammesDbToDomain/PromotionMapper.php

```diff
@@ -32,14 +32,12 @@ public function getDomainModel(array $dbPromotion): Promotion
      */
     private function getCacheKeyForPromotion(array $dbPromotion): string
     {
-        return $this->buildCacheKey(
-            $dbPromotion,
-            'id',
-            [
-                'promotionOfCoreEntity' => 'CoreEntity',
-                'promotionOfImage' => 'Image',
-            ]
-        );
+        return $this->buildCacheKey($dbPromotion, 'id', [
+            'promotionOfCoreEntity' => 'CoreEntity',
+            'promotionOfImage' => 'Image',
+        ], [
+            'relatedLinks' => 'RelatedLink',
```

Comment:
Why the separate array here specifying this as a many/many relationship? Isn't this oneToMany?

Reply:
We're returning an array of related links in the `relatedLinks` key, not a single one.

The first array in building the cache keys is for ManyToOne relationships - where the promotionOfImage field relates to a single entity.
The second array in building the cache keys if for OneToMany relationships - where the relatedLinks field is an array.

The logic for building the cache keys our for those two scenarios is slightly different as one is `1` and the other is `[1,2]`